templates/assets/env-ecs.yml

```diff
@@ -400,6 +404,8 @@ Resources:
           /opt/aws/bin/cfn-init -v --stack ${AWS::StackName} --resource ContainerInstances --configsets ${ImageOsType} --region ${AWS::Region} $CFN_PROXY_ARGS
           /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackName} --resource EcsAutoScalingGroup --region ${AWS::Region} $CFN_PROXY_ARGS
 
+          ${ExtraUserData}
```

Comment:
This should go above the `cfn-signal` command to ensure CloudFormation waits for the extra user data to run successfully